Projects/Core/Services/Haptic/HapticClient.swift

```diff
@@ -0,0 +1,43 @@
+//
+//  HapticClient.swift
+//  Services
+//
+//  Created by GREEN on 6/8/24.
+//  Copyright © 2024 com.mashup.gabbangzip. All rights reserved.
+//
+
+import ComposableArchitecture
+import SwiftUI
+
+// MARK: - API Client Interface
+public struct HapticClient {
+  /// 사용자 인터랙션에 따른 햅틱 반응 (light || medium || heavy || soft || rigid)
+  public var triggerImpact: @Sendable (UIImpactFeedbackGenerator.FeedbackStyle) async -> Void
+  /// 작업 완료 여부에 따른 햅틱 반응 (success || warning || error)
+  public var triggerNotification: @Sendable (UINotificationFeedbackGenerator.FeedbackType) async -> Void
+}
+
+public extension DependencyValues {
+  var hapticClient: HapticClient {
+    get { self[HapticClient.self] }
+    set { self[HapticClient.self] = newValue }
+  }
+}
+
+// MARK: - API Client Implementation
+extension HapticClient: DependencyKey {
+  public static let liveValue = HapticClient(
+    triggerImpact: { style in
+      await MainActor.run {
+        let generator = UIImpactFeedbackGenerator(style: style)
+        generator.impactOccurred()
+      }
+    },
+    triggerNotification: { type in
+      await MainActor.run {
+        let generator = UINotificationFeedbackGenerator()
+        generator.notificationOccurred(type)
+      }
+    }
```

Comment:
```suggestion
    triggerImpact: { style in
      let generator = await UIImpactFeedbackGenerator(style: style)
      await generator.impactOccurred()
    },
    triggerNotification: { type in
      let generator = await UINotificationFeedbackGenerator()
      await generator.notificationOccurred(type)
    }
```
`MainActor`의 `run()` 메서드 대신 await 키워드만으로도 적용이 가능합니다. 저는 개인적으로 await만 사용하는 것이 더 가독성이 좋아보이는데 어떻게 생각하시나요?

Reply:
오 햅틱은 UI 관련 API라 메인스레드 보장을 하도록 명시적으로 가져왔는데, TCA의 리듀서에서 사용 시 .run이나 .fireAndForget으로 감싸 메인 스레드로 보장하는 식의 사용이 더 적절할까요?

Reply:
`UINotificationFeedbackGenerator`, `UIImpactFeedbackGenerator`에 `@MainActor